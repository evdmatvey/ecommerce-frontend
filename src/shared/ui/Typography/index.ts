export { default } from './ui/Typography';
