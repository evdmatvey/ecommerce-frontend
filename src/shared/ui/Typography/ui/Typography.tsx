import clsx from 'clsx';
import { ReactNode } from 'react';
import styles from './Typography.module.css';

interface TypographyProps {
  component?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span' | 'div';
  variant?:
    | 'display 1'
    | 'display 2'
    | 'display 3'
    | 'display 4'
    | 'display 5'
    | 'display 6'
    | 'heading 1'
    | 'heading 2'
    | 'heading 3'
    | 'heading 4'
    | 'heading 5'
    | 'heading 6'
    | 'xl'
    | 'l'
    | 'nav link'
    | 'nav link s'
    | 'regular'
    | 's'
    | 'xs'
    | 'nav link xs';
  children: ReactNode;
}

const Typography = ({
  component: Tag = 'div',
  variant = 'regular',
  children,
}: TypographyProps) => {
  const classnames = clsx({
    [styles.display1]: variant === 'display 1',
    [styles.display2]: variant === 'display 2',
    [styles.display3]: variant === 'display 3',
    [styles.display4]: variant === 'display 4',
    [styles.display5]: variant === 'display 5',
    [styles.display6]: variant === 'display 6',
    [styles.heading1]: variant === 'heading 1',
    [styles.heading2]: variant === 'heading 2',
    [styles.heading3]: variant === 'heading 3',
    [styles.heading4]: variant === 'heading 4',
    [styles.heading5]: variant === 'heading 5',
    [styles.heading6]: variant === 'heading 6',
    [styles.xl]: variant === 'xl',
    [styles.l]: variant === 'l',
    [styles.navLink]: variant === 'nav link',
    [styles.navLinkS]: variant === 'nav link s',
    [styles.navLinkXs]: variant === 'nav link xs',
    [styles.s]: variant === 's',
    [styles.xs]: variant === 'xs',
  });
  return <Tag className={classnames}>{children}</Tag>;
};

export default Typography;
