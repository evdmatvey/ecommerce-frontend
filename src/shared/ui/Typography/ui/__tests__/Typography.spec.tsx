import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Typography from '../Typography';

const TEST_TEST = 'Test text';

describe('Typography', () => {
  it('Standard should render correctly', () => {
    const { asFragment } = render(<Typography>{TEST_TEST}</Typography>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Display should render correctly', () => {
    const { asFragment } = render(
      <Typography variant="display 1">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Heading should render correctly', () => {
    const { asFragment } = render(
      <Typography variant="heading 1">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Large should render correctly', () => {
    const { asFragment } = render(
      <Typography variant="l">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Nav link should render correctly', () => {
    const { asFragment } = render(
      <Typography variant="nav link">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Small should render correctly', () => {
    const { asFragment } = render(
      <Typography variant="s">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('H1 should render correctly', () => {
    const { asFragment } = render(
      <Typography component="h1">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('P should render correctly', () => {
    const { asFragment } = render(
      <Typography component="p">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Span should render correctly', () => {
    const { asFragment } = render(
      <Typography component="span">{TEST_TEST}</Typography>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
