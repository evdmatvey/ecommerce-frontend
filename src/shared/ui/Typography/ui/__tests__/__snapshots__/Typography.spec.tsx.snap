// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Typography > Display should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_display1_4a62f9"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`Typography > H1 should render correctly 1`] = `
<DocumentFragment>
  <h1
    class=""
  >
    Test text
  </h1>
</DocumentFragment>
`;

exports[`Typography > Heading should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_heading1_4a62f9"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`Typography > Large should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_l_4a62f9"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`Typography > Nav link should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_navLink_4a62f9"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`Typography > P should render correctly 1`] = `
<DocumentFragment>
  <p
    class=""
  >
    Test text
  </p>
</DocumentFragment>
`;

exports[`Typography > Small should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_s_4a62f9"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`Typography > Span should render correctly 1`] = `
<DocumentFragment>
  <span
    class=""
  >
    Test text
  </span>
</DocumentFragment>
`;

exports[`Typography > Standard should render correctly 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    Test text
  </div>
</DocumentFragment>
`;
